flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/utils/ParquetSchemaConverter.java

```diff
@@ -306,6 +295,18 @@ public static MessageType toParquetType(TypeInformation<?> typeInformation, bool
 		return typeInfo;
 	}
 
+	private static ObjectArrayTypeInfo convertGroupElementToArrayTypeInfo(GroupType arrayFieldType, GroupType elementType) {
+		for (Type type : elementType.getFields()) {
+			if (!type.isRepetition(Type.Repetition.REQUIRED)) {
+				throw new UnsupportedOperationException(
```

Comment:
Why this check? What's wrong with having a list of complex elements, whose fields are not all required?